modules/Microsoft.DigitalTwins/digitalTwinsInstances/endpoints-eventGrid/deploy.bicep

```diff
@@ -0,0 +1,64 @@
+@description('Optional. The name of the Digital Twin Endpoint.')
+param name string = 'EventGridEndpoint'
+
+@description('Conditional. The name of the parent Digital Twin Instance resource. Required if the template is used in a standalone deployment.')
+param digitalTwinInstanceName string
+
+@description('Required. EventGrid Topic Endpoint.')
+param topicEndpoint string
+
+@description('Required. Event Grid Resource Id.')
+param eventGridDomainId string
+
+@description('Optional. Dead letter storage secret for key-based authentication. Will be obfuscated during read.')
+@secure()
+param deadLetterSecret string = ''
+
+@description('Optional. Dead letter storage URL for identity-based authentication.')
+param deadLetterUri string = ''
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+var eventGridAccessKey1 = listkeys(eventGridDomainId, '2022-06-15').key1
+
+var eventGridAccessKey2 = listkeys(eventGridDomainId, '2022-06-15').key2
+
```

Comment:
```suggestion
```

I'd recommend to fetch them directly in the parameters block as they are not used anywhere else.